site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+kind: Gateway
+metadata:
+  name: gateway-b
+spec:
+  gatewayClassName: nadir-lb
+  listeners:
+  - name: http
+    protocol: HTTP
+    port: 80
+    hostname: subdomain.example.com
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-parent
+spec:
+  parentRefs:
+  - name: gateway-a
+    sectionName: http
+  - name: gateway-b
+    sectionName: http
+  rules:
+    - allowedRoutes:
+        kinds: 
+          - kind: HTTPRoute
+        namespaces:
+          from: Same
+    - backendRefs:
+      - name: default-svc
+        port: 8080
+status:
+  parents:
+  - controllerName: acme
+    parentRef:
+      name: gateway-a
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      message: Successfully accepted
+  - controllerName: nadir
+    parentRef:
+      name: gateway-b
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      message: Successfully accepted
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-child
+spec:
+  parentRefs:
+  - name: gateway-a
+    sectionName: http
+  hostnames:
+    - testdomain.example.com
+  rules:
+    - backendRefs:
+      - name: foo-svc
+        port: 8080
+status:
+  parents:
+  - controllerName: acme
+    parentRef:
+      name: http-parent
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      message: Accepted successfully
+    grandparents:
+      - parentRef:
+          name: gateway-a
+        conditions:
+        - type: Accepted
+          status: true
+          reason: Accepted
+          message: Successfully accepted
+  - controllerName: nadir
+    parentRef:
+      name: http-parent
+    conditions:
+    - type: Accepted
+      status: true
+      reason: GrandparentConfigError
+      message: Some configuration was not accepted in the attachment process. See grandparents for more details.
+    grandparents:
+      - parentRef:
+          name: gateway-b
+        conditions:
+        - type: Accepted
+          status: false
+          reason: HostnameMismatch
+          message: The route could not be accepted because of a hostname mismatch - `testdomain.example.com` does not match `subdomain.example.com`.
+```
+
+
+
+## Alternatives
+
+### Extending backendRef to allow targeting a Route
+
+This idea was to allow `backendRef` to, optionally, specify a route instead of a
+service. Some of the reasons for not taking this approach include:
+
+- It would require changes to `backendRefs`. No changes to `allowedRoutes` are
+  needed to use it for this. 
+- To minimize the changes to `backendRef`, it would require the owner of the
+  parent route to know the exact name of the child route. If the two routes are
+  in different namespaces, that might require cross-organization coordination
+  between administrators.
+- It would result in there being one way to attach routes to listeners and a
```

Comment:
Maybe from an implementation perspective. From a user perspective a route can be seen as a backend -- its logically going through both of the routes, they just happen to be merged together as part of the implementation details (of some implementations! some may not actually do this at all).

Reply:
The other thing I don't think we mentioned here is that I think that the backendRef implies an extra routing hop, which I don't think will be common. This is a way to break up the config, not a way to represent extra routing hops.

Reply:
Why would backendRef 'imply' an extra routing hop ? It allows implementations to use an extra hop, if strong isolation between parent/child is required - or to flatten the route. 

Technically Service is not a real backend and does require CNI to route requests to a Pod, and the spec allows any CRD
to be used as backend - there is no mention of 'hops' in the spec.